api/controllers/CampaignBotController.js

```diff
@@ -104,24 +104,27 @@ CampaignBotController.prototype.chatbot = function(req, res) {
 CampaignBotController.prototype.createUserAndPostSignup = function(req, res) {
   var self = this;
 
-  var newUser = {
-    _id: req.user_id,
-    mobile: req.user_mobile,
-    campaigns: {}
-  };
+  app.locals.northstarClient.Users.get('id', req.user_id).then(function(user) {
```

Comment:
Just a general suggestion: instead of thening `dbUsers.create()` model inside `Users.get.then()`, you can instead return from it `dbUsers.create()` and `then()` the result one level above. Thus both thens are handled by the `catch` at the end. Something like this:

```js

app.locals.northstarClient.Users.get('id', req.user_id)
  .then(function(user) {
    const userFields = {
      _id: req.user_id,
      mobile: req.user_mobile,
      first_name: user.firstName,
      email: user.email,
      phoenix_id: user.drupalID,
      campaigns: {},
    };
    return dbUsers.create(userFields);
  }
  .then(function (userDoc) {
    self.user = userDoc;
    logger.debug('%s created user', self.loggerPrefix(req));
    self.postSignup(req, res);

  })
  .catch(function(err) {self.handleError(req, res, err)});
```